app/src/renderer/vuex/modules/delegation.js

```diff
@@ -66,14 +66,14 @@ export default ({ commit }) => {
       await dispatch('walletTx', args)
     },
     async submitDelegation ({ state, dispatch }, delegation) {
-      for (let delegate of delegation.delegates) {
+      return Promise.all(delegation.delegates.map(delegate => {
```

Comment:
Running these in parallel does not help since we have to sequentially increment the nonce and check that the tx went through. Really this should all happen in an atomic operation, but the LCD does not support that. Will open a ticket.